### PR TITLE
NXP-29966: compute video info, previews, storyboard and conversions only when needed

### PR DESCRIPTION
**Pull request by troger**
When creating a document (new document or version), we compute:
- the video info only if the width and height is set to 0
- the video previews only if empty
- the video storyboard only if empty
- only the missing automatic conversions not present on the document

When updating the main blob document, everything is reset and recomputed.

### CODE REVIEW

File: modules/platform/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/VideoServiceImpl.java [hunk 1]
```diff
@@ -148,10 +149,18 @@ public void launchConversion(DocumentModel doc, String conversionName) {
 
     @Override
     public void launchAutomaticConversions(DocumentModel doc) {
+        launchAutomaticConversions(doc, false);
```
Comment (efge) [inline modules/platform/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/VideoServiceImpl.java:152]: Could you put this in the interface as a default method instead?

File: modules/platform/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/VideoStoryboardWork.java [hunk 2]
```diff
@@ -81,28 +85,51 @@ public void work() {
...
         // update storyboard
         setStatus("Updating storyboard");
-        log.debug(String.format("Updating storyboard of Video document %s.", doc));
-        VideoHelper.updateStoryboard(doc, video);
-        log.debug(String.format("End updating storyboard of Video document %s.", doc));
+        boolean save = updateStoryboard(doc, video);
 
         // update previews
         setStatus("Updating previews");
+        save = updatePreviews(doc, video) || save;
```
Comment (efge) [inline modules/platform/nuxeo-platform-video/src/main/java/org/nuxeo/ecm/platform/video/service/VideoStoryboardWork.java:92]: `save |= updatePreviews(doc, video);`

**Review by efge — CHANGES_REQUESTED** [review]

**Review by efge — APPROVED** [review]

**Review by kevinleturc — DISMISSED** [review]

File: modules/platform/nuxeo-platform-video/src/test/java/org/nuxeo/ecm/platform/video/extension/TestVideoImporterAndListeners.java [hunk 3]
```diff
@@ -482,4 +487,294 @@ public void testNoVideoConversionWhenDisabled() throws IOException {
...
+        List<Map<String, Serializable>> pictureviews = new ArrayList<>();
+        Map<String, Serializable> pictureView = new HashMap<>();
+        pictureView.put("title", "StaticPlayerView");
+        pictureView.put("content", (Serializable) Blobs.createBlob("dummy"));
+        pictureviews.add(pictureView);
+        doc.setPropertyValue("picture:views", (Serializable) pictureviews);
+        doc = session.createDocument(doc);
+        txFeature.nextTransaction();
+
+        // only video info and previews computed
```
Comment (ataillefer) [inline modules/platform/nuxeo-platform-video/src/test/java/org/nuxeo/ecm/platform/video/extension/TestVideoImporterAndListeners.java:726]:
```suggestion
        // only video info and previews are set
```
as they actually haven't been computed by the listener

**Review by ataillefer — APPROVED** [review]

**Comment by troger** [comment]
Updated following @ataillefer's comment.

**Review by kevinleturc — APPROVED** [review]

**Review by efge — APPROVED** [review]

**Review by ataillefer — APPROVED** [review]

**Pull request merged**